### PR TITLE
feat(user-onboarding): base code for user-onboarding lib

### PR DESCRIPTION
**Pull request by vaibhavmauryasf**
## Description
Base code of the user-onboarding client
Fixes # (issue)
https://github.com/sourcefuse/loopback4-microservice-catalog/issues/173
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [x] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/user-onboarding/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# UserOnboarding
```
Comment (akshatdubeysf) [inline packages/user-onboarding/README.md:1]: Add Readme on how to use, etc.

File: packages/user-onboarding/package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (akshatdubeysf) [inline packages/user-onboarding/package.json:1]: change it to something like @sourceloop/user-onboarding-client

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  constructor(private readonly tourStoreService: TourStoreServiceService) {}
+
+  private triggerTour(tourInstance: Tour){
+    var removedSteps;
+    const tour = new Shepherd.Tour({
+      defaultStepOptions: {
+        cancelIcon: {
+          enabled: true
+        },
+        classes: 'class-1 class-2',
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts:21]: are these required?

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+          });
+        });
+        tour.addSteps(removedSteps);
+        for(const step of removedSteps)
+        {
+            tour.steps.splice(0,0,tour.steps.pop());
+        }
+      }
+      tour.on("show",()=>{
+        this.tourStoreService.saveState({tourId: tourInstance.tourId,state:{sessionId: this.tourStoreService.getSessionId(),step: tour.getCurrentStep().id}});
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts:73]: Looks like the code is not prettified, use something like [this](https://www.npmjs.com/package/prettier).

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  private readonly defaultLTCommand = new LoadTCommand(this.storage);
+
+  constructor(@Inject(LOCAL_STORAGE) private readonly storage: StorageService) {
+      this.commandMap.set("SaveTourCommand",this.defaultSTCommand);
+      this.commandMap.set("LoadTourCommand",this.defaultLTCommand);
+      this.commandMap.set("SaveStateCommand",this.defaultSSCommand);
+      this.commandMap.set("LoadStateCommand",this.defaultLSCommand);
+   }
+  registerSaveTourCommand(cmd){
+    this.commandMap.delete("SaveTourCommand");
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts:26]: I don't think you need this line, as you are setting a value on this key just after this line.

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/user-onboarding-lib.component.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/user-onboarding-lib.component.ts:1]: This is not needed.

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/user-onboarding-lib.service.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@angular/core';
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/lib/user-onboarding-lib.service.ts:1]: ??

File: packages/user-onboarding/projects/user-onboarding-lib/src/public-api.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Public API Surface of user-onboarding-lib
+ */
+export * from './lib/user-onboarding-lib.service';
+export * from './lib/user-onboarding-lib.component';
+export * from './lib/user-onboarding-lib.module';
+export * from './lib/tour-service.service';
+export * from './lib/tour-store-service.service';
+export * from './models/StateParameters';
```
Comment (akshatdubeysf) [inline packages/user-onboarding/projects/user-onboarding-lib/src/public-api.ts:9]: You don't need to export individual models if you have an index file in the models folder.

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

**Comment by akshatdubeysf** [comment]
Also add a link to the user-onboarding issue.

**Comment by akshatdubeysf** [comment]
Fix linting issues.

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts (hunk 3 above)
Comment (vaibhavmauryasf) [reply packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-service.service.ts:21]: yes sir, like if someone wants to cancel the tour in between so there's a cross icon for that that's why cancelIcon is used and classes are regarding the theming of the tour

File: packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts (hunk 5 above)
Comment (vaibhavmauryasf) [reply packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts:26]: the line with the delete statement?

Comment (akshatdubeysf) [reply packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts]: Yes

Comment (vaibhavmauryasf) [reply packages/user-onboarding/projects/user-onboarding-lib/src/lib/tour-store-service.service.ts]: already fixed sir

**Comment by akshatdubeysf** [comment]
Fixed spelling mistakes in readme.

**Review by akshatdubeysf — APPROVED** [review]

**Review by akshatdubeysf — APPROVED** [review]

**Pull request merged**